plugins/net_plugin/peer_keys_db.cpp

```diff
@@ -0,0 +1,46 @@
+#include <eosio/net_plugin/peer_keys_db.hpp>
+#include <eosio/chain/contract_table_objects.hpp>
+
+namespace eosio::chain {
+
+peer_keys_db_t::peer_keys_db_t()
+   : _peer_key_map( boost::shared_ptr<peer_key_map_t>(new peer_key_map_t)) {
+}
+
+void peer_keys_db_t::on_accepted_block(const controller& chain) {
+   try {
+      const auto& head = chain.head();
+
+      // let's update once every 20 blocks, so every 10 seconds
+      if (head.is_valid() && block_header::num_from_id(head.id()) % 20 == 0) {
+         auto new_map = boost::shared_ptr<peer_key_map_t>(new peer_key_map_t);
```

Comment:
`std::shared_ptr` and might as well use `make_shared`.

Reply:
I think I have to use `boost_shared_ptr` to initialize a `boost::atomic_shared_ptr`, but I'll use `boost::make_shared`.